atlasdb-dbkvs-hikari/src/main/java/com/palantir/nexus/db/pool/config/OracleConnectionConfig.java

```diff
@@ -72,12 +74,17 @@ public String getUrl() {
     @Value.Derived
     @JsonIgnore
     public Optional<String> namespace() {
-        if (getSid().isPresent()) {
-            return getSid();
-        }
-        return serviceNameConfiguration().map(ServiceNameConfiguration::namespaceOverride);
+        return Stream.of(
+                        namespaceOverride(),
+                        getSid(),
+                        serviceNameConfiguration().map(ServiceNameConfiguration::namespaceOverride))
+                .filter(Optional::isPresent)
+                .findFirst()
+                .orElseThrow(() -> new SafeIllegalStateException("Could not determine namespace for Oracle config"));
     }
 
+    public abstract Optional<String> namespaceOverride();
```

Comment:
`JsonProperty("namespace")`? Or do we want `oracleNamespace`, to explicitly be different from the existing/C* config?

Reply:
Yep, no harm doing that. I think I went with `override` for explicit precedence order, though it's a fair argument saying it's implicit.